src/tss2-fapi/api/Fapi_AuthorizePolicy.c

```diff
@@ -0,0 +1,284 @@
+ * @retval TSS2_FAPI_RC_BAD_REFERENCE: if context, policyPath or keyPath
+ *         is NULL.
+ * @retval TSS2_FAPI_RC_BAD_CONTEXT: if context corruption is detected.
+ * @retval TSS2_FAPI_RC_BAD_PATH: if policyPath or keyPath does not
+ *         map to a FAPI policy or key object.
+ * @retval TSS2_FAPI_RC_BAD_SEQUENCE: if the context has an asynchronous
+ *         operation already pending.
+ * @retval TSS2_FAPI_RC_IO_ERROR: if the data cannot be saved.
+ * @retval TSS2_FAPI_RC_MEMORY: if the FAPI cannot allocate enough memory for
+ *         internal operations or return parameters.
+ */
+TSS2_RC
+Fapi_AuthorizePolicy_Async(
+    FAPI_CONTEXT  *context,
+    char    const *policyPath,
+    char    const *keyPath,
+    uint8_t const *policyRef,
+    size_t         policyRefSize)
+{
+    TSS2_RC r;
+    IFAPI_Fapi_AuthorizePolicy *policy;
+
+    r = ifapi_session_init(context);
+    return_if_error(r, "Initialize AuthorizePolicy");
+
+    if (!policyPath) {
+        return_error(TSS2_FAPI_RC_BAD_REFERENCE,
+                     "Must not be NULL: policyPath.");
+    }
+
+    policy = &context->cmd.Policy_AuthorizeNewPolicy;
+    policy->policyPath = policyPath;
+    policy->signingKeyPath = keyPath;
+    if (policyRef) {
+        FAPI_COPY_DIGEST(&policy->policyRef.buffer[0],
+                         policy->policyRef.size, policyRef, policyRefSize);
+    } else {
+        policy->policyRef.size = 0;
+    }
+    r = ifapi_session_init(context);
+    return_if_error(r, "Initialize PolicyAuthorizeNewPolicy");
+
+    context->state = AUTHORIZE_NEW_LOAD_KEY;
+
+    return r;
+}
+
+/** Asynchronous finish function for Fapi_AuthorizePolicy
+ *
+ * This function should be called after a previous Fapi_AuthorizePolicy_Async.
+ *
+ * @param [in, out] context The FAPI_CONTEXT
+ *
+ * @retval TSS2_RC_SUCCESS: if the function call was a success.
+ * @retval TSS2_FAPI_RC_BAD_REFERENCE: if context is NULL.
+ * @retval TSS2_FAPI_RC_BAD_CONTEXT: if context corruption is detected.
+ * @retval TSS2_FAPI_RC_BAD_SEQUENCE: if the context has an asynchronous
+ *         operation already pending.
+ * @retval TSS2_FAPI_RC_IO_ERROR: if the data cannot be saved.
+ * @retval TSS2_FAPI_RC_MEMORY: if the FAPI cannot allocate enough memory for
+ *         internal operations or return parameters.
+ * @retval TSS2_FAPI_RC_TRY_AGAIN: if the asynchronous operation is not yet
+ *         complete. Call this function again later.
+ */
+TSS2_RC
+Fapi_AuthorizePolicy_Finish(
+    FAPI_CONTEXT *context)
+{
+    TSS2_RC r;
+    TPMI_ALG_HASH hash_alg;
+    IFAPI_CRYPTO_CONTEXT_BLOB *cryptoContext;
+    size_t hash_size;
+    size_t digest_idx;
+    IFAPI_Fapi_AuthorizePolicy *policy =
+        &context->cmd.Policy_AuthorizeNewPolicy;
+    TPMS_POLICYAUTHORIZATION *authorization = &policy->authorization;
+    TPMT_SIGNATURE *signature;
+    TPM2B_DIGEST aHash;
+    char *publicKey = NULL;
+
+    switch (context->state) {
+    case AUTHORIZE_NEW_LOAD_KEY:
```

Comment:
In some cases we use the statecase macro and in some cases, like here, just case. Any reason for the inconsistency?

Reply:
statecase should now be everywhere